bgpd/bgp_route.c

```diff
@@ -2090,6 +2108,28 @@ bool subgroup_announce_check(struct bgp_dest *dest, struct bgp_path_info *pi,
 					   p, &label);
 			return false;
 		}
+	} else if (safi == SAFI_MPLS_VPN && from && peer && pi->peer != peer &&
+		   pi->sub_type != BGP_ROUTE_IMPORTED &&
+		   pi->sub_type != BGP_ROUTE_STATIC &&
+		   bgp_mplsvpn_path_uses_valid_mpls_label(pi) &&
+		   bgp_path_info_nexthop_changed(pi, peer, afi) &&
+		   pi->mplsvpn.bmnc.nh_label_bind_cache) {
```

Comment:
you do not check what value in union is used

Reply:
Please move that `pi->mplsvpn.bmnc.nh_label_bind_cache` just after `safi == ` test
If the feature is not enabled, no need to test further

Reply:
done

Reply:
OK